File: examples/example-53/ruby/features/support/messaging.rb
require 'bunny'
require './src/messaging'

messaging_start

at_exit do
    messaging_stop
end

Before('@rabbitmq') do
    @connection = Bunny.new
    @connection.start
    @channel = @connection.create_channel
end

After('@rabbitmq') { @connection.close }

File: examples/example-53/js/features/support/messaging.js
const amqp = require('amqplib');
const { Before, After, BeforeAll, AfterAll } = require('@cucumber/cucumber');
const { connectMessaging, closeMessaging } = require('../../src/messaging');

BeforeAll(async () => {
    await connectMessaging();
});

AfterAll(async () => {
    await closeMessaging();
});

Before({ tags: '@rabbitmq' }, async function hook() {
    this.amqp = { connection: await amqp.connect('amqp://localhost') };
    this.amqp.channel = await this.amqp.connection.createChannel();
});

After({ tags: '@rabbitmq' }, async function hook() {
    if (!this.amqp?.connection) { return; }
    await this.amqp.connection.close();
});